### ISSUE TITLE
Most of the theme colors are broken for me

### ISSUE THREAD

**Issue by ahmedelgabri** [syl20bnr/spacemacs#1269]
Here is a screenshot trying to use `base16-ocean` theme

![screen shot 2015-04-20 at 00 07 57](https://cloud.githubusercontent.com/assets/63876/7221603/6b02ac7c-e6f1-11e4-91fb-97371d90078d.png)

Monokai is the only one that is working nicely, how can I fix this? I'm using iTerm2

**Comment by trishume** [comment]
You can fix it by using the GUI version. Most themes are borked in terminal, don't know why.

**Comment by EricTing** [comment]
put `export TERM=xterm-256color` in your `~/.bashrc` or `~/.profile` and try.

**Comment by ahmedelgabri** [comment]
@EricTing My terminal is already set as `xterm-256color`

**Comment by syl20bnr** [comment]
@ahmedelgabri can you open a helm-color buffer with `SPC C l` and tell us how many colors you see there ?

**Comment by ahmedelgabri** [comment]
@syl20bnr I can see 100 colors

**Comment by tuhdo** [comment]
It should be 100 because Helm is limited to display 100 candidates. However, it means that you have 256 colors. Maybe you should give Zenburn a try? Or use stock Terminal app in OS X rather than Iterm 2 to see if there's any difference?

**Comment by ahmedelgabri** [comment]
@tuhdo The same is happening on the terminal.app. Only Zenburn _I don't like it_ and Monokai are the only ones that are working fine most of the rest if not all are broken.

**Comment by tuhdo** [comment]
Even Solarized look ugly? So I think probably your terminal and Emacs works fine, it's just that the themes are not well supported in terminal. Monokai actually defines another set of colors for terminal version.

**Comment by ahmedelgabri** [comment]
@tuhdo Yeah even Solarized. Other than this everything looks like working well.

**Comment by Baltazore** [comment]
Does anyone know how to fix this issue? Any suggestions? 
I already tried: ```TERM=xterm-256color emacs```

**Comment by honza** [comment]
![screen shot 2015-05-18 at 09 00 36](https://cloud.githubusercontent.com/assets/206357/7680382/69998f34-fd3c-11e4-8cad-dfbc7e3de007.png)

This is my setup with solarized dark in iTerm2 and tmux.

**Comment by bobbyangelov** [comment]
I have this issue as well! Tried re-installing emacs and that didn't work. For now I guess we're stuck with the GUI version.

**Comment by madmod** [comment]
Same issue here in both iTerm 2 and Terminal. Also the powerline is messed up even when I use a powerline patched font. I've set my iTerm 2 font to Source Code Pro with the "Report Terminal Type" option on all possible settings. I tried running emacs-24.5 also with the commands suggested above. Using the GUI version isn't a great option for me.

**Comment by cpaulik** [comment]
In the terminal you can only get one of the 256 colors listed [here](http://en.wikipedia.org/wiki/File:Xterm_256color_chart.svg). This means that if the theme does not take care the colors set by the theme will go to the nearest color in terms of distance in color space.

AFAIK the only solution is to choose a theme with terminal support.

**Comment by Baltazore** [comment]
@cpaulik is there any port of solarized vim terminal theme for emacs?

**Comment by cpaulik** [comment]
I don't know why but the themes-megapack uses http://melpa.org/#/color-theme-sanityinc-solarized which does not seem to support color degradation properly.

http://melpa.org/#/color-theme-solarized on the other hand seems to handle this issue [well](https://github.com/sellout/emacs-color-theme-solarized/blob/master/solarized-definitions.el#L70-L90)

@syl20bnr Any reason for choosing the one over the other?

**Comment by madmod** [comment]
It looks like someone [got 24 bit color working](http://www.reddit.com/r/vim/comments/2ozwe4/24_bit_vim_in_osx_iterm2_a_reality/) with vim using a nightly build of iTerm 2. I don't know how to adapt this to spacemacs though.

**Comment by trishume** [comment]
That's neat I didn't know iTerm supported that. There's 24 bit patched formulae for tmux and emacs here:
https://github.com/choppsv1/homebrew-term24

**Comment by ahmedelgabri** [comment]
I'm already using iTerm nightly and that screenshot was from an iTerm nightly build.

**Comment by ahmedelgabri** [comment]
Although it does support more colors and that fixed my issues when I use vim with base-16 but apparently not working with Emacs.

**Comment by syl20bnr** [comment]
@cpaulik the solarized version used in spacemacs is [bbatsov's one](https://github.com/bbatsov/solarized-emacs).

I just realized that we could create a new dotvariable for spacemacs `dotspacemacs-terminal-themes` !!

**Comment by trishume** [comment]
@ahmedelgabri you have to use both the right iTerm and a patched emacs. See the link I posted.

**Comment by cpaulik** [comment]
@syl20bnr ah ok. From a quick glance at the source code of this version it seems that it also does not have terminal support. 

So we could add [this version of solarized](http://melpa.org/#/color-theme-solarized) to the themes-megapack but how many different solarized themes do we want to have in spacemacs?

**Comment by madmod** [comment]
Why does the current solarized theme work in the GUI emacs but not with true color in iTerm 2? Can we patch emacs or spacemacs so that it thinks iTerm is the GUI?

-- 
John Wells


On June 10, 2015 at 2:30:25 PM, Christoph Paulik (notifications@github.com) wrote:

@syl20bnr ah ok. From a quick glance at the source code of this version it seems that it also does not have terminal support.

So we could add this version of solarized to the themes-megapack but how many different solarized themes do we want to have in spacemacs?

**Comment by trishume** [comment]
@madmod Did you see the link I posted to a patched emacs that does exactly that?

**Comment by ahmedelgabri** [comment]
@trishume was on my phone so I didn't check the link. But why do we need to patch Emacs while it seems _as other mentioned_ that the problem might be the themes themselves?

**Comment by madmod** [comment]
@trishume sorry I missed that. After running the following commands in the latest nightly build of iTerm 2 I have 24 bit color working with the sublime-dark theme!

![image](https://cloud.githubusercontent.com/assets/1107818/8094934/5f568088-0f87-11e5-8350-1d2f0bb4e7f6.png)

First I had to uninstall my previous version of emacs and then install the patched version.
```bash
brew uninstall emacs
brew tap choppsv1/term24
brew install choppsv1/term24
```

Also for anyone interested this is a fast way to install the nightly version of iTerm2. By default casks will install to `~/Applications`. Run install with `--appdir` if you want it elsewhere. [Cask usage](https://github.com/caskroom/homebrew-cask/blob/master/USAGE.md)
```bash
brew tap caskroom/versions
brew cask install iterm2-nightly
```

Then run emacs with the flag [mentioned here].(https://gist.github.com/choppsv1/e8198df70cb1778b04d6/#file-emacs-24-5-24bit-diff-L7-L14)
```bash
ITERM_24BIT=1 emacs
```

The last thing I need to figure out is how to fix the bad characters at the bottom. I have iTerm set to use Source Code Pro for both the ASCII and non ASCII font.

Other minor things not working for me:
- The cursor is not colored like in GUI emacs.
- There is no highlight on mouseover for clickable items.

**Comment by trishume** [comment]
@madmod the characters in the bottom you need a powerline patched font for.

**Comment by VMatrix1900** [comment]
same problem here, both iTerm2 and Terminal.

**Comment by jasonsmith0** [comment]
Following, how to workaround the solarized issue and have a working solarized dark in terminal?

**Comment by jhegedus42** [comment]
@cpaulik How can I use this http://melpa.org/#/color-theme-solarized with spacemacs atm ?

**Comment by cpaulik** [comment]
@jhegedus42 What is the difference between the one you posted and http://melpa.org/#/solarized-theme ? Because http://melpa.org/#/solarized-theme is as easy as adding `solarized-light` or `solarized-dark` to the `dotspacemacs-themes` list.

**Comment by cpaulik** [comment]
Oh, I'm sorry it's about the terminal colors isn't it.

**Comment by jhegedus42** [comment]
If i just use solarized-dark : then I get => https://files.gitter.im/syl20bnr/spacemacs/GsY3/Screen-Shot-2015-10-11-at-11.32.47.png

**Comment by d12frosted** [comment]
I thought colours depends on your **terminal** theme. So if you want to use solarized theme in emacs - use it for terminal as well.

**Comment by cpaulik** [comment]
In an unchanged terminal the theme degrades to one of the [256 default colors](https://en.wikipedia.org/wiki/File:Xterm_256color_chart.svg). You can however change your terminal settings to get the exact colors. This is also what is the recommended solution of the color-theme-solarized https://github.com/sellout/emacs-color-theme-solarized#important-note-for-terminal-users

Using color-theme-solarized should work if you exclude all the other solarized packages and load the solarized theme in your `user-config`

**Comment by jhegedus42** [comment]
Probably i don't understand something - i have been using the solarized mode in vim and tmux and mc pretty happily, i just don't get why it does not work in emacs. ? no idea

**Comment by StreakyCobra** [comment]
@cpaulik's last comment explain well what the issue with terminal colors. It should be documented in the FAQ.

**Comment by jrhorn424** [comment]
@cpaulik Forgive me for being dense. It's been a while since I used emacs, and even though I clicked through I'm a bit confused.

 My terminal scheme is solarized-dark. My spacemacs theme is solarized-dark, and I have color issues.

Should I download an `.Xdefaults` to get spacemacs-in-terminal to work? Or should disable themes?

**Comment by cpaulik** [comment]
Other themes should work fine. So if you can find one you like besides solarized then this would be the easiest solution at the moment.

You can also change the default 256 colors of your terminal emulator following https://github.com/sellout/emacs-color-theme-solarized#important-note-for-terminal-users.

**Comment by morgado-jb** [comment]
Changing the default 256 colors of the terminal doesn't fix the problem.

I'm using base16 both with iterm2 and with emacs and practically all the themes (not only solarized) show this bug. I use the same method in Vim and there it works fine (although only after using this method of putting the same base16 scheme both in vim and in iterm2), so there is something more at play here with emacs.

**Comment by jhegedus42** [comment]
Just my two cents: plain emacs with evil-mode works fine with solarized colors in iterm2 out of the box. Spacemacs does not ( for me).

**Comment by morgado-jb** [comment]
Reason for closing the bug?

**Comment by StreakyCobra** [comment]
It was marked as `Fixed in develop`. Those one are closed when a new release is out. This issue is borderline, so I'm reopening.

**Comment by cpaulik** [comment]
I'm not sure that this is really solvable in spacemacs. If somebody has set a 'emacs' colortheme that does not care about degradation to 256 colors then it is unlikely to look good in a standard terminal. 

For the people using a terminal theme I think that they have to use the standard emacs color theme for it to look correct. So in this case there should be an option in spacemacs to not set a theme. Or is there an emacs-default theme on MELPA?

**Comment by TheBB** [comment]
No, you are right, we do not handle the default theme correctly.

**Comment by morgado-jb** [comment]
Every color theme I've used shows this behaviour, including 2 major points: The default theme and the base-16 emacs theme (which is rather constrained in its functionality, so even if it worked it wouldn't be a good help) even when setting the corresponding base-16 shell theme.

In OSX the problem is mitigated by using the 24bit patch for homebrew Emacs (any news on it being pushed to homebrew base Casks?).

In Linux, under the shell it makes Spacemacs unusable.

**Comment by cpaulik** [comment]
@morgado-jb I use spacemacs in a linux terminal without major issues. Of course the theme has to care about setting colors that look sane.

Also, if I understand things correctly then you should not set a emacs color theme *and* a shell theme but only one of the two. Otherwise the colors that the emacs themes sets will be remapped by the shell theme which seems unlikely to yield correct results.

Some themes I tried (all with gnome terminal emulator)

spacemacs-dark
![image](https://cloud.githubusercontent.com/assets/380927/12091455/91ffb7c0-b2f7-11e5-898e-bf5f94e59c4e.png)

monokai
![image](https://cloud.githubusercontent.com/assets/380927/12091672/b1f7643c-b2f8-11e5-88bb-f40572490fa0.png)


material theme
![image](https://cloud.githubusercontent.com/assets/380927/12091441/66736890-b2f7-11e5-8cf4-347224b165d1.png)

**Comment by JacobJakub** [comment]
OMG @madmod thanks a lot for the commands. Finally could use a blue theme! looks beautiful now

**Comment by qiuwei** [comment]
@jhegedus42 
Yes, I have exactly the same problem. Solarized-theme  works for vim, vanilla emacs, iterm2, but fails for spaceemacs.

Have you found any workaround?

**Comment by robbyoconnor** [comment]
solarized in spacemacs looks like utter crap. I've been using monokai.

**Comment by morgado-jb** [comment]
Monokai works mostly ok except for the highlighting. Colors are a bit off, but it doesn't look awful like Solarized.

Still, it would be nice to solve the problem, either by finding a nice color theme (or 2, one dark, one light) that works properly in the TTY.

I still can't understand why Vim works perfectly both in Linux and OSX if I set up the console and Vim to use the Solarized theme, but Emacs and spacemacs doesn't. It's really an impediment for people that can't/don't want to use the graphical Emacs.

**Comment by cpaulik** [comment]
@morgado-jb So which themes are you using at which level? I'm asking because using:

- emacs: solarized
- terminal emulator: solarized

Can not work AFAIK.

What should work is

- emacs: default
- terminal emulator: solarized

You have to take care of the correct mapping of the colors you have set in emacs to the colors in your terminal. The [default 256 xterm colors](https://upload.wikimedia.org/wikipedia/en/1/15/Xterm_256color_chart.svg) do not map well to solarized.

**Comment by dpc** [comment]
I'm a emacs noob. Is there any workaround for that or not? I'm trying switch from Vim, but I'm a heavy terminal user. I have everything (gnome-terminal, tmux, vim, and other stuff) set to work with solarized and they work just fine.

**Comment by robbyoconnor** [comment]
Why not use the GUI -- you get so much more with it. Plus with`pdf-tools` layer in `develop` -- PDFs are amazing now.

**Comment by cpaulik** [comment]
@dpc I've done some research and I think that at the moment there is no solarized theme package that is compatible with spacemacs and the terminal out of the box. But there is a workaround (see bottom of this post)

- The [official one](https://github.com/sellout/emacs-color-theme-solarized) uses a loading mechanism that is not compatible with spacemacs and also has no clear distinction between light and dark theme but it has to be set manually
- The [solarized theme](https://github.com/bbatsov/solarized-emacs) on MELPA has [an issue](https://github.com/bbatsov/solarized-emacs/issues/18) for the problem 
- The [sanityinc version of purcell](https://github.com/purcell/color-theme-sanityinc-solarized) also has an [issue](https://github.com/purcell/color-theme-sanityinc-solarized/issues/16) for this problem.

I've also tested this now since this solarized discussion really bugs me. The only way I found to get this working was to configure the terminal to use the 16 solarized colors by using e.g. [this for gnome terminal](https://github.com/Anthony25/gnome-terminal-colors-solarized). You can then add `color-theme-solarized` to your `dotspacemacs-additional-packages`. Then add

```
  (set-terminal-parameter nil 'background-mode 'light)
  (set-frame-parameter nil 'background-mode 'light)
  (spacemacs/load-theme 'solarized)
```

to your `dotspacemacs/user-config`. This should enable the solarized-light theme for both terminal and GUI. If you want dark then change the background mode above

**Comment by portOdin** [comment]
Could someone tell me the Spacemacs way to change the background colour of a theme? Or maybe how to remove the colour all together, so that my iTerm background colour shows through? It's strange that the only colour that isn't correct in a lot of these themes is the background. Though at least that should be an easy fix.

**Comment by portOdin** [comment]
Thanks a lot @madmod, I got it working. But the theme is completely missing when I run Emacs in Tmux. Anyone know what might be causing this?

Without tmux:

![screen shot 2016-03-18 at 8 56 22 pm](https://cloud.githubusercontent.com/assets/5838941/13895634/229620ce-ed4c-11e5-8627-1c92c2cf03f9.png)

With tmux:

![screen shot 2016-03-18 at 8 57 50 pm](https://cloud.githubusercontent.com/assets/5838941/13895636/2b5509d2-ed4c-11e5-80db-87ff0990f1a4.png)

**Comment by dpc** [comment]
@portOdin : What does your $SHELL is to when you're in tmux? Try adding `set-option -g default-terminal screen-256color` to your `~/.tmux.conf`.

**Comment by dpc** [comment]
@cpaulik : It worked. I already had the solarized gnome-terminal colorscheme, and I just had modify `~/.spacemacs`. Thanks. I hope the fix will land in standard theme, but for now I'm OK.

**Comment by ethanluoyc** [comment]
Actually I think the idea of having a separate variable for terminal theme is nice. @syl20bnr do you have plans to implement that?

**Comment by syl20bnr** [comment]
@ethanluoyc unfortunately I don't think this is possible because it will break with Emacs started as a daemon.

**Comment by dpc** [comment]
My currently best working setup (solarized dark in terminal, through ssh, in tmux, using daemon + client): http://termbin.com/groe

**Comment by robbyoconnor** [comment]
Why not just use TRAMP vs ssh'ing in?

**Comment by ethanluoyc** [comment]
The problem is that Tramp can have lots of bugs sometimes.





On Thu, Jul 7, 2016 at 4:25 PM +0800, "robbyoconnor" <notifications@github.com> wrote:












Why not just use TRAMP vs ssh'ing in?

**Comment by dpc** [comment]
@robbyoconnor: I am personally looking for Spacemacs to be my Vim replacement. I'd like to be able to take a cheap chromebook, and ssh to my dev box and use Emacs instead of Vim, instead of trying to install emacs on my chromebook etc.

The fact that command line is a second-class citizen in emacs is a great pain for me, since I already spent so much time altering my habits, I can clearly see that emacs is more powerful than Vim, and I really like Spacemacs features and SPC leader, but then problems with solarized theme, and display corruptions in tmux are ruining the experience. :(

**Comment by abnerf** [comment]
@cpaulik your fix works beautifully, but not with emacs as daemon and connecting with emacsclient. Do you have any idea of how to fix it?

**Comment by cpaulik** [comment]
@abnerf have you tried https://github.com/sellout/emacs-color-theme-solarized#all-versions ? I guess you want to set the theme when a new frame is created.

**Comment by dpc** [comment]
@abnerf : I use setup with combination of  gnome-shell with dark solarized, tmux, ssh, spacemacs in client + daemon mode and it works for me quite well now.

* http://sprunge.us/RJZL - `.spacemacs`
* http://sprunge.us/HSPN - command to start emacs deamon
* http://sprunge.us/ceGT - command to start emacs

**Comment by abnerf** [comment]
@cpaulik @dpc thanks! I had solved with a similar solution. Now, one thing I don't understand is why the font size is different/smaller when I launch the emacsclient (with graphics) compared to the normal emacs (with graphics). But that probably a question for another ticket.

**Comment by arronmabrey** [comment]
@domi91c did you ever figure out why tmux was pure black / white? I'm having the same issue.

**Comment by yogsototh** [comment]
The best workaround I found it to add this line in my `.spacemacs` in the function `dotspacemacs/user-config`:

~~~elisp
  ;; Colortheme fix in terminal
  (custom-set-faces (if (not window-system) '(default ((t (:background "nil"))))))
~~~

So now I am happy with my `alias vi="emacsclient -t"`.

**Comment by markschultz** [comment]
@yogsototh I've expanded on that a bit to allow `emacsclient -t` and `emacsclient -c`

```
 (add-hook 'after-make-frame-functions
            (lambda (frame)
              (select-frame frame)
              (if (not window-system)
                  (custom-set-faces '(default ((t (:background "nil")))))
                  (custom-set-faces '(default ((t (background dark))))))))

```
using urxvt on linux I was getting correct colors in gui, dark blue background in terminal. Above gives more or less correct colors in both.

**Comment by robbyoconnor** [comment]
@markschultz -- do a PR to fix this for others -- or document it someplace...

**Comment by bezhermoso** [comment]
@yogsototh's solution works to fix colors in iTerm, but eventually breaks `Emacs.app` once the `(custom-set-faces ...)` is appended to `~/.spacemacs`.

@markschultz's solution doesn't fix it for me.

**Comment by DKuykendall** [comment]
Yogsototh's solution worked for me, but in both the terminal and gui, the background of the mode/buffer number indicator on the left of the powerline is now no longer colored, with the exception of the powerline separator character. No other theme does this. Any idea what it could be?
<img width="160" alt="screen shot 2017-01-26 at 2 45 04 am" src="https://cloud.githubusercontent.com/assets/8590650/22323530/963e7dc0-e371-11e6-9a37-d4200976a8f0.png">
<img width="185" alt="screen shot 2017-01-26 at 2 45 17 am" src="https://cloud.githubusercontent.com/assets/8590650/22323532/97f9e6ea-e371-11e6-82e7-1188e2e8a130.png">

**Comment by belak** [comment]
I know that there was work done on the base16-themes to default to using your terminal colors, so the should no longer be an issue with this.

**Comment by jupl** [comment]
For those using tmux the answer [here](http://askubuntu.com/a/350604) fixed my weird color issues.

TL;DR `TERM=screen-256color` and `tmux -2 [other options]`

**Comment by fmthoma** [comment]
So, I finally got @sellout's [emacs-color-theme-solarized](https://github.com/sellout/emacs-color-theme-solarized) to work with the `emacs` daemon and both GUI and terminal (`urxvt`) clients.

These are the steps:

## Step 1:
Set up a Solarized color scheme for your terminal. For urxvt, the `.Xresources` can be found e.g. [here](https://github.com/solarized/xresources). I added a semi-transparent background: `scratchpad.background: [80]#002b36`.

## Step 2:
In your `.spacemacs`, you have to exclude the default package for the Solarized color theme (which is broken in most terminals), and install the other one:

```elisp
(defun dotspacemacs/layers ()
  (setq-default
   ...
   dotspacemacs-excluded-packages '(solarized-theme)
   dotspacemacs-additional-packages '(color-theme-solarized)
   ...
   ))
```
And by the way, you should also remove any `solarized-dark`/`solarized-light` from your `dotspacemacs-themes`.

## Step 3:
The theme loading mechanism for `color-theme-solarized` is somewhat different from the Spacemacs default, so we need a bit of tweaking. Add this to your `dotspacemacs/user-config`:

```elisp
;; For GUI clients, after-make-frame-functions is messed up (probably it is
;; called too early), and some colors are messed up. For terminal clients on
;; the other hand, it works perfectly. The background-mode is necessary for
;; terminal clients, but (apparently) not for GUI clients, and has to be set
;; for each frame individually.
(spacemacs|do-after-display-system-init (enable-solarized-in-gui))
(add-hook 'after-make-frame-functions 'enable-solarized-in-terminal)

(defun enable-solarized-in-terminal (frame)
  ;; cf. http://philipdaniels.com/blog/2017/02/spacemacs---configuring-the-solarized-theme/
  (unless (display-graphic-p frame)
    (set-frame-parameter frame 'background-mode 'dark)
    (set-terminal-parameter frame 'background-mode 'dark)
    (spacemacs/load-theme 'solarized)
    ))

(defun enable-solarized-in-gui ()
  (mapc 'disable-theme custom-enabled-themes)
  (setup-solarized-theme)
  (spacemacs/load-theme 'solarized)
  )

(defun setup-solarized-theme ()
  ;; I like a transparent background in the terminal
  (custom-set-faces
    '(default (
              (((type tty) (background dark)) (:background "nil"))
              )))
  ;; The frame number indicator has some problems with inverse video
  (set-face-inverse-video 'spacemacs-motion-face nil)
  (set-face-inverse-video 'spacemacs-insert-face nil)
  (set-face-inverse-video 'spacemacs-normal-face nil)
  (set-face-inverse-video 'spacemacs-visual-face nil)
  (set-face-inverse-video 'spacemacs-replace-face nil)
  )
```

## Explanation
The difference to @yogsototh's and @markschultz's approach for disabling the background for terminal clients is basically not to change the `'default` face definition each time a frame is created, but to assign it a frame-dependent value: the `(type tty) (background dark)` in the `custom-set-faces` call means: »When the `frame-type` is `tty`, and the `background-mode` is `dark`, then change the `:background` to `"nil"`.
The `set-face-inverse-video` lines solve the problem mentioned by @DKuykendall.
Also (and this took me a lot of time to figure out), the two different hooks are important: Apparently, in daemon mode the `after-make-frame-functions` hook is called too early for GUI clients, so the display system is not yet fully initialized and some colors remain broken. (They magically fix after starting the second client, and the problem does neither appear for tty clients, nor for standalone instances).

**Comment by fiveNinePlusR** [comment]
Is this still an issue?

**Comment by belak** [comment]
@fiveNinePlusR This needs to be fixed on a theme-by-theme basis. The base16 themes are the only ones I can directly speak for, but they should be fine.

**Comment by mrivas616** [comment]
@fmthoma Your solution worked for me, not only I got solarized to finally look the way it's supposed to look but also any other theme looks good now in GUI mode, thank you!

**Comment by martinweiss** [comment]
I am not using spacemacs, but I also had issues with bbatsov's Solarized themes when running emacs in the terminal. After a while of searching for possible solutions (which also led me here), I finally found a solution that worked for me, by using a custom terminfo description `(xterm-24bit)`, as desribed here: https://github.com/bbatsov/solarized-emacs/issues/18#issuecomment-294374266

Now I no longer get that blue background when I use the solarized-dark theme.

Maybe this also works with spacemacs?

**Comment by sdwolfz** [comment]
There's a guide for turning on 24bit in a terminal here: https://github.com/syl20bnr/spacemacs/wiki/Terminal

**Comment by mengfankitty** [comment]
@sdwolfz It works for me!! I'm using ir-black theme.

**Comment by Compro-Prasad** [comment]
IMO this issue is resolved through the wiki.
